### PR TITLE
recipe detail product  customization

### PR DESCRIPTION
**Pull request by ThibaultMiam**

### CODE REVIEW

**Comment by ThibaultMiam** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/26?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#29** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/29?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#28** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/28?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#27** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/27?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#26** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/26?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#25** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/25?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#24** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/24?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#23** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/23?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#22** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/22?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#21** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/21?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#20** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/20?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#19** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/19?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#18** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/18?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#17** <a href="https://app.graphite.dev/github/pr/miamtech/mealz-documentation/17?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `develop`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @ThibaultMiam and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx [hunk 1]
```diff
@@ -0,0 +1,635 @@
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+### product
```
Comment (diarmuidmcg) [inline docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx:4]:
```suggestion
### Product
```

File: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx [hunk 2]
```diff
@@ -0,0 +1,635 @@
...
+
+Product has three sub-templates : 
+- **success** : state when data is loaded from back end
+- **loading** : still fetching data 
+- **ignored** : product won't be fetched 
+
```
Comment (diarmuidmcg) [inline docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx:10]:
```suggestion
- **success** : State when data is loaded from back end
- **loading** : Still fetching data 
- **ignored** : Product will not be shown or added to basket (yet ingredient is still shown)

```

File: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx [hunk 3]
```diff
@@ -0,0 +1,635 @@
...
+    }
+    ignore {
+        view = MyCustomRecipeDetailProductIgnore()
+    }
+}
+```
+
+#### success
+
+To create your own recipe detail product success template you have to create class that implement `ProductSuccess`
```
Comment (diarmuidmcg) [inline docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx:27]:
```suggestion
#### Success

To create your own recipe detail product success template you create a class that implements `ProductSuccess`
```

File: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx [hunk 4]
```diff
@@ -0,0 +1,635 @@
...
+    val replaceProduct: () -> Unit,
+    val ignoreProduct: () -> Unit,
+    val addProduct: () -> Unit,
+    val updateProductQuantity: (Int) -> Unit
+)
+```
+
+#### loading
+
+To create your own recipe detail product loading template you have to create class that implement `ProductLoading`
```
Comment (diarmuidmcg) [inline docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx:337]:
```suggestion
#### Loading

To create your own recipe detail product loading template you create a class that implements `ProductLoading`
```

File: docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx [hunk 5]
```diff
@@ -0,0 +1,635 @@
...
+
+with 
+
+```kotlin
+object ProductLoadingParameters
+```
+
+#### ignored 
+
+To create your own recipe detail product ignore template you have to create class that implement `ProductIgnore`
```
Comment (diarmuidmcg) [inline docs/android/features/_shared/_custom-views/recipeDetail/RecipeDetailProduct.mdx:511]:
```suggestion
#### Ignored 

To create your own recipe detail product ignore template you create a class that implements `ProductIgnore`
```

**Review by diarmuidmcg — CHANGES_REQUESTED** [review]
I would have put the components into sep files so its easy to manage

**Pull request merged**